### PR TITLE
AWS Configuration Validation and Logging

### PR DESCRIPTION
**Pull request by AdamBien**
Validates AWS-specific configuration and writes additional log statements

### CODE REVIEW

File: implementation/jwt-auth/src/main/java/io/smallrye/jwt/auth/principal/AwsAlbKeyConfigurationValidator.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            throw AwsAlbKeyResolverMessages.msg.subPathNotAllowed();
+        }
+    }
+
+    public static void validatePublicKeyAlgorithmConfiguration(JWTAuthContextInfo authContextInfo) {
+        var publicKeyAlgorithm = authContextInfo.getSignatureAlgorithm();
+        if (publicKeyAlgorithm == null) {
+            AwsAlbKeyResolverLogging.log.publicKeyAlgorithmNotSet();
+        }
+        if (!publicKeyAlgorithm.getAlgorithm().equals("ES256")) {
```
Comment (sberyozkin) [inline implementation/jwt-auth/src/main/java/io/smallrye/jwt/auth/principal/AwsAlbKeyConfigurationValidator.java:25]: Please consider `SignatureAlgorithms.ES256.getAlgorithm()`

**Comment by sberyozkin** [comment]
Thanks @AdamBien :+1: for improving the logging related to AWS ALB, interesting that no sub paths are allowed in the key location URL. LGTM, minor suggestion added to use `SignatureAlgorithm` enum.

@Skyllarr, please check early next week if all looks good to you

Cheers

**Comment by AdamBien** [comment]
Hey Sergey,

you can only specify the root path. The key path is dynamically added from the kid. 

cheers,

adam

> On 1. Oct 2023, at 21:24, Sergey Beryozkin ***@***.***> wrote:
> 
> 
> Thanks @AdamBien 👍 for improving the logging related to AWS ALB, interesting that no sub paths are allowed in the key location URL. LGTM, minor suggestion added to use SignatureAlgorithm enum.
> @Skyllarr, please check early next week if all looks good to you
> Cheers
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you were mentioned.Message ID: ***@***.***>

 about.adam-bien.com
 airhacks.industries

**Comment by sberyozkin** [comment]
Hey @AdamBien 

>you can only specify the root path. The key path is dynamically added from the kid.

OK, so that will add the only path segment to the root path.

Thanks

File: implementation/jwt-auth/src/main/java/io/smallrye/jwt/auth/principal/AwsAlbKeyConfigurationValidator.java (hunk 1 above)
Comment (sberyozkin) [reply implementation/jwt-auth/src/main/java/io/smallrye/jwt/auth/principal/AwsAlbKeyConfigurationValidator.java:25]: @AdamBien I've prepared a tiny update to resolve this suggestion, are you OK if I push to your branch ?

**Comment by AdamBien** [comment]
sure!

**Review by sberyozkin — APPROVED** [review]

**Comment by sberyozkin** [comment]
@Skyllarr FYI, this PR improves how AWS ALB key resolver validates its configuration and logs various issues encountered during the validation. I'll merge tomorrow unless you have some additional comments, thanks

**Pull request merged**